projects/plugins/boost/app/modules/optimizations/page-cache/pre-wordpress/boost-cache-actions.php

```diff
@@ -71,6 +72,32 @@ function jetpack_boost_delete_cache_for_url( $url ) {
  * @param int $post_id - The ID of the post to delete the cache for.
  */
 function jetpack_boost_delete_cache_by_post_id( $post_id ) {
+	$post = get_post( (int) $post_id );
+	if ( ! $post ) {
+		return;
+	}
+	static $already_deleted = -1;
+	if ( $already_deleted === $post->ID ) {
+		return;
+	}
+
+	/**
+	 * Don't invalidate the cache for post types that are not public.
+	 */
+	if ( ! Boost_Cache_Utils::is_visible_post_type( $post ) ) {
+		return;
+	}
+
+	$already_deleted = $post->ID;
+
+	/**
+	 * If a post is unpublished, the permalink will be deleted. In that case,
+	 * get_sample_permalink() will return a permalink with ?p=123 instead of
+	 * the post name. We need to get the post name from the post object.
+	 */
+	$permalink = get_permalink( $post->ID );
+	Logger::debug( "invalidate_cache_for_post: $permalink" );
+
 	$boost_cache = new Boost_Cache();
-	$boost_cache->invalidate_cache_by_post_id( (int) $post_id, Filesystem_Utils::DELETE_ALL );
+	$boost_cache->delete_page( $permalink );
```

Comment:
I guess this should be `delete_recursive` as well.

Reply:
Yeah. A post may have paged comments, so yeah. This is now using delete_post_cache which does a recursive delete.